lib/ansible/plugins/action/__init__.py

```diff
@@ -1104,7 +1104,10 @@ def _get_diff_data(self, destination, source, task_vars, source_file=True):
         display.debug("Going to peek to see if file has changed permissions")
         peek_result = self._execute_module(module_name='file', module_args=dict(path=destination, _diff_peek=True), task_vars=task_vars, persist_files=True)
 
-        if not peek_result.get('failed', False) or peek_result.get('rc', 0) == 0:
+        if peek_result.get('failed', False):
+            raise AnsibleError(u"Failed to get diff between '%s' and '%s': %s" % (os.path.basename(source), destination, to_text(peek_result.get(u'msg', u''))))
```

Comment:
Should this be an error?  My opinion is that a diff is informative only.  Perhaps if there is a failure a warning instead, and just disable diff for the task?

Reply:
Agreed. I'll change it to a warning instead.